### PR TITLE
sql-statements: flashback cluster support through ddl history

### PR DESCRIPTION
**Pull request by Defined2014**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 1]
```diff
@@ -33,9 +33,10 @@ FlashbackToTimestampStmt ::=
     ```
 
 * 执行 `FLASHBACK CLUSTER` SQL 语句的用户需要有 `SUPER` 权限。
-* 在 `FLASHBACK` 指定的时间点到开始执行的时间段内不能存在相关表结构变更的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
+* `FLASHBACK CLUSTER` 指定的时间点不能存在未执行完成的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
 * 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
 * `FLASHBACK CLUSTER TO TIMESTAMP` 命令不能取消，一旦开始执行 TiDB 会一直重试，直到成功。
+* 若执行 `FLASHBACK CLUSTER` 的集群创建了 TiCDC 同步任务，需要用户手动将下游的schema 定义进行恢复，并在 `FLASHBACK` 完成后重新创建 changefeed。
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-to-timestamp.md:39]:
```suggestion
* 若执行 `FLASHBACK CLUSTER` 的集群创建了 TiCDC 同步任务，需要用户手动将下游的 schema 定义进行恢复，并在 `FLASHBACK CLUSTER` 完成后重新创建 changefeed。
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 2]
```diff
@@ -74,22 +75,22 @@ mysql> SELECT * FROM t;
 Empty set (0.00 sec)
 ```
 
-如果从 `FLASHBACK` 指定的时间点到开始执行的时间段内有改变表结构的 DDL 记录，那么将执行失败：
+如果 `FLASHBACK CLUSTER` 指定的时间点有未完成的 DDL 记录，那么将执行失败：
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-to-timestamp.md:78]:
```suggestion
如果 `FLASHBACK CLUSTER` 指定的时间点有未完成的 DDL 记录，那么 `FLASHBACK CLUSTER` 将执行失败：
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 3]
```diff
@@ -74,22 +75,22 @@ mysql> SELECT * FROM t;
...
+
+mysql> admin show ddl jobs 1;
++--------+---------+-----------------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
+| JOB_ID | DB_NAME | TABLE_NAME            | JOB_TYPE               | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | CREATE_TIME         | START_TIME          | END_TIME            | STATE  |
++--------+---------+-----------------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
+|     84 | test    | t                     | add index /* ingest */ | public       |         2 |       82 |         0 | 2023-01-29 14:33:11 | 2023-01-29 14:33:11 | 2023-01-29 14:33:12 | synced |
++--------+---------+-----------------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
+1 rows in set (0.01 sec)
+
+mysql> flashback cluster to timestamp '2023-01-29 14:33:12';
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-to-timestamp.md:92]:
```suggestion
mysql> ALTER TABLE t ADD INDEX k(a);
Query OK, 0 rows affected (0.56 sec)

mysql> ADMIN SHOW DDL JOBS 1;
+--------+---------+-----------------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
| JOB_ID | DB_NAME | TABLE_NAME            | JOB_TYPE               | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | CREATE_TIME         | START_TIME          | END_TIME            | STATE  |
+--------+---------+-----------------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
|     84 | test    | t                     | add index /* ingest */ | public       |         2 |       82 |         0 | 2023-01-29 14:33:11 | 2023-01-29 14:33:11 | 2023-01-29 14:33:12 | synced |
+--------+---------+-----------------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
1 rows in set (0.01 sec)

mysql> FLASHBACK CLUSTER TO TIMESTAMP '2023-01-29 14:33:12';
```

**Review by ran-huang — COMMENTED** [review]
Minor comments. Rest LGTM

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 4]
```diff
@@ -33,9 +33,11 @@ FlashbackToTimestampStmt ::=
     ```
 
 * 执行 `FLASHBACK CLUSTER` SQL 语句的用户需要有 `SUPER` 权限。
-* 在 `FLASHBACK` 指定的时间点到开始执行的时间段内不能存在相关表结构变更的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
-* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
+* `FLASHBACK CLUSTER` 不支持回退修改 PD 相关信息的 DDL, 如 `ALTER TABLE ATTRIBUTE`, `ALTER TABLE REPLICA`, `CREATE PLACEMENT POLICY` 等。
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-to-timestamp.md:36]:
```suggestion
* `FLASHBACK CLUSTER` 不支持回退修改 PD 相关信息的 DDL，如 `ALTER TABLE ATTRIBUTE`、`ALTER TABLE REPLICA`、`CREATE PLACEMENT POLICY` 等。
```

**Review by ran-huang — APPROVED** [review]

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 5]
```diff
@@ -33,9 +33,11 @@ FlashbackToTimestampStmt ::=
     ```
 
 * 执行 `FLASHBACK CLUSTER` SQL 语句的用户需要有 `SUPER` 权限。
-* 在 `FLASHBACK` 指定的时间点到开始执行的时间段内不能存在相关表结构变更的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
-* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
+* `FLASHBACK CLUSTER` 不支持回退修改 PD 相关信息的 DDL，如 `ALTER TABLE ATTRIBUTE`、`ALTER TABLE REPLICA`、`CREATE PLACEMENT POLICY` 等。
+* `FLASHBACK CLUSTER` 指定的时间点不能存在未执行完成的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
+* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK CLUSTER` 完成。
 * `FLASHBACK CLUSTER TO TIMESTAMP` 命令不能取消，一旦开始执行 TiDB 会一直重试，直到成功。
+* 若执行 `FLASHBACK CLUSTER` 的集群创建了 TiCDC 同步任务，需要用户手动将下游的 schema 定义进行恢复，并在 `FLASHBACK CLUSTER` 完成后重新创建 changefeed。
```
Comment (TomShawn) [inline sql-statements/sql-statement-flashback-to-timestamp.md:40]:
```suggestion
* 若在执行 `FLASHBACK CLUSTER` 的集群上创建了 TiCDC 同步任务，需要用户手动将下游的 schema 定义进行恢复，并在 `FLASHBACK CLUSTER` 完成后重新创建 changefeed。
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 6]
```diff
@@ -74,22 +76,22 @@ mysql> SELECT * FROM t;
...
++--------+---------+-----------------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
+|     84 | test    | t                     | add index /* ingest */ | public       |         2 |       82 |         0 | 2023-01-29 14:33:11 | 2023-01-29 14:33:11 | 2023-01-29 14:33:12 | synced |
++--------+---------+-----------------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
+1 rows in set (0.01 sec)
+
+mysql> FLASHBACK CLUSTER TO TIMESTAMP '2023-01-29 14:33:12';
+ERROR 1105 (HY000): Detected another DDL job at 2023-01-29 14:33:12 +0800 CST, can't do flashback
 ```
 
 可以通过日志查看 `FLASHBACK` 执行进度，具体的日志如下所示：
```
Comment (TomShawn) [inline sql-statements/sql-statement-flashback-to-timestamp.md:97]: 日志的时间是否需要相应地更新一下呢？

**Review by TomShawn — APPROVED** [review]
rest LGTM

File: sql-statements/sql-statement-flashback-to-timestamp.md (hunk 6 above)
Comment (Defined2014) [reply sql-statements/sql-statement-flashback-to-timestamp.md:97]: 日志是单独的，和前面失败的例子没有关联性

File: sql-statements/sql-statement-flashback-to-timestamp.md (hunk 5 above)
Comment (Defined2014) [reply sql-statements/sql-statement-flashback-to-timestamp.md:40]: 感觉加了 `若在执行` 之后语义有点变了，`若在`感觉有点正执行的意思。

Comment (Defined2014) [inline sql-statements/sql-statement-flashback-to-timestamp.md:40]:
```suggestion
* 若 `FLASHBACK CLUSTER` 导致了元信息（表结构、库结构）的回滚，则元信息的回滚 ** 不会 ** 被 TiCDC 同步。因此，用户需主动暂停任务，待 `FLASHBACK CLUSTER` 完成后将上下游的 schema 定义手动同步一致，然后重新创建 TiCDC changefeed。
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 7]
```diff
@@ -33,9 +33,11 @@ FlashbackToTimestampStmt ::=
     ```
 
 * 执行 `FLASHBACK CLUSTER` SQL 语句的用户需要有 `SUPER` 权限。
-* 在 `FLASHBACK` 指定的时间点到开始执行的时间段内不能存在相关表结构变更的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
-* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
+* `FLASHBACK CLUSTER` 不支持回退修改 PD 相关信息的 DDL，如 `ALTER TABLE ATTRIBUTE`、`ALTER TABLE REPLICA`、`CREATE PLACEMENT POLICY` 等。
+* `FLASHBACK CLUSTER` 指定的时间点不能存在未执行完成的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
+* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK CLUSTER` 完成。
 * `FLASHBACK CLUSTER TO TIMESTAMP` 命令不能取消，一旦开始执行 TiDB 会一直重试，直到成功。
+* 若 `FLASHBACK CLUSTER` 导致了元信息（表结构、库结构）的回滚，则元信息的回滚 ** 不会 ** 被 TiCDC 同步。因此，用户需主动暂停任务，待 `FLASHBACK CLUSTER` 完成后将上下游的 schema 定义手动同步一致，然后重新创建 TiCDC changefeed。
```
Comment (Defined2014) [inline sql-statements/sql-statement-flashback-to-timestamp.md:40]:
```suggestion
* 若 `FLASHBACK CLUSTER` 导致了元信息（表结构、库结构）的回滚，则相关的修改 ** 不会 ** 被 TiCDC 同步。因此，用户需主动暂停任务，待 `FLASHBACK CLUSTER` 完成后将上下游的 schema 定义手动同步一致，然后重新创建 TiCDC changefeed。
```

**Review by bb7133 — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 8]
```diff
@@ -33,9 +33,11 @@ FlashbackToTimestampStmt ::=
     ```
 
 * 执行 `FLASHBACK CLUSTER` SQL 语句的用户需要有 `SUPER` 权限。
-* 在 `FLASHBACK` 指定的时间点到开始执行的时间段内不能存在相关表结构变更的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
-* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
+* `FLASHBACK CLUSTER` 不支持回退修改 PD 相关信息的 DDL，如 `ALTER TABLE ATTRIBUTE`、`ALTER TABLE REPLICA`、`CREATE PLACEMENT POLICY` 等。
+* `FLASHBACK CLUSTER` 指定的时间点不能存在未执行完成的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
+* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK CLUSTER` 完成。
 * `FLASHBACK CLUSTER TO TIMESTAMP` 命令不能取消，一旦开始执行 TiDB 会一直重试，直到成功。
+* 若 `FLASHBACK CLUSTER` 导致了元信息（表结构、库结构）的回滚，则相关的修改 ** 不会 ** 被 TiCDC 同步。因此，用户需主动暂停任务，待 `FLASHBACK CLUSTER` 完成后将上下游的 schema 定义手动同步一致，然后重新创建 TiCDC changefeed。
```
Comment (Defined2014) [inline sql-statements/sql-statement-flashback-to-timestamp.md:40]:
```suggestion
* 若 `FLASHBACK CLUSTER` 导致了元信息（表结构、库结构）的回滚，则相关的修改 **不会** 被 TiCDC 同步。因此，用户需主动暂停任务，待 `FLASHBACK CLUSTER` 完成后将上下游的 schema 定义手动同步一致，然后重新创建 TiCDC changefeed。
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 9]
```diff
@@ -33,9 +33,11 @@ FlashbackToTimestampStmt ::=
     ```
 
 * 执行 `FLASHBACK CLUSTER` SQL 语句的用户需要有 `SUPER` 权限。
-* 在 `FLASHBACK` 指定的时间点到开始执行的时间段内不能存在相关表结构变更的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
-* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
+* `FLASHBACK CLUSTER` 不支持回退修改 PD 相关信息的 DDL，如 `ALTER TABLE ATTRIBUTE`、`ALTER TABLE REPLICA`、`CREATE PLACEMENT POLICY` 等。
+* `FLASHBACK CLUSTER` 指定的时间点不能存在未执行完成的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
+* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK CLUSTER` 完成。
 * `FLASHBACK CLUSTER TO TIMESTAMP` 命令不能取消，一旦开始执行 TiDB 会一直重试，直到成功。
+* 若 `FLASHBACK CLUSTER` 导致了元信息（表结构、库结构）的回滚，则相关的修改 **不会** 被 TiCDC 同步。因此，用户需主动暂停任务，待 `FLASHBACK CLUSTER` 完成后将上下游的 schema 定义手动同步一致，然后重新创建 TiCDC changefeed。
```
Comment (Defined2014) [inline sql-statements/sql-statement-flashback-to-timestamp.md:40]:
```suggestion
* 若 `FLASHBACK CLUSTER` 导致了元信息（表结构、库结构）的回滚，则相关的修改**不会**被 TiCDC 同步。因此，用户需主动暂停任务，待 `FLASHBACK CLUSTER` 完成后将上下游的 schema 定义手动同步一致，然后重新创建 TiCDC changefeed。
```

**Pull request merged**